File: favodemel-api/src/FavoDeMel.Domain/Entities/Comandas/Comanda.cs
using FavoDeMel.Domain.Common;
using FavoDeMel.Domain.Entities.Usuarios;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace FavoDeMel.Domain.Entities.Comandas
{
    public class Comanda : Entity<Guid>
    {
        public decimal TotalAPagar { get; set; }
        public decimal GorjetaGarcom { get; set; }
        public string Codigo { get; set; }
        public DateTimeOffset DataCadastro { get; set; }
        public ComandaSituacao Situacao { get; set; }

        public Usuario Garcom { get; set; }
        [ForeignKey("ComandaId")]
        public ICollection<ComandaPedido> Pedidos { get; set; }

        public Comanda()
        {
            Pedidos = new Collection<ComandaPedido>();
        }

        public void FecharConta()
        {
            TotalAPagar = Pedidos.Sum(c => c.Quantidade * c.Produto.Preco);
            GorjetaGarcom = Garcom != null ? (Garcom.Comissao / 100) * TotalAPagar : 0;
            Situacao = ComandaSituacao.Fechada;
        }

        public void Confirmar()
        {
            Situacao = ComandaSituacao.EmAndamento;
        }
    }
}

File: favodemel-web/src/app/models/comanda.ts
export class Comanda {
  id: any;
  garcomId: any;
  pedidos: ComandaPedido[];
  totalAPagar: any;
  gorjetaGarcom: any;
  situacao: ComandaSituacao;
  dataCadastro: any;
  codigo: string;
}

export class ComandaPedido {
  id: any;
  produtoId: any;
  produtoNome: string;
  produtoPreco: any;
  quantidade: number;
  situacao: ComandaPedidoSituacao;
}

export enum ComandaSituacao {
  Aberta = 1,
  EmAndamento = 2,
  Fechada = 3
}

export enum ComandaPedidoSituacao {
  Pedido = 1,
  Preparando = 2,
  Pronto = 3
}
